astropy/io/fits/tests/test_core.py

```diff
@@ -1286,6 +1288,32 @@ def test_simulateonly(self):
             hdul[0].writeto(None)
             hdul[0].header.tofile(None)
 
+    def test_bintablehdu_zero_bytes(self):
+        """Make sure we don't have any zero-byte writes in BinTableHDU"""
+
+        bright = np.rec.array([(1, 'Sirius', -1.45, 'A1V'),
+                            (2, 'Canopus', -0.73, 'F0Ib'),
+                            (3, 'Rigil Kent', -0.1, 'G2V')],
+                           formats='int16,a20,float32,a10', names='order,name,mag,Sp')
+
+        hdu_non_zero = fits.BinTableHDU(bright)
+        # use safeio, a special file handler meant to fail on zero-byte writes
+        fh = safeio.CatchZeroByteWriter(open('bright.fits', mode='wb'))
+        hdu_non_zero.writeto(fh)
+        fh.close()
+
+    def test_primaryhdu_zero_bytes(self):
+        """
+        Make sure we don't have any zero-byte writes from an ImageHDU
+        `(or other) of `size % BLOCK_SIZE == 0`
```

Comment:
```suggestion
        (or other) of `size % BLOCK_SIZE == 0`
```
stray backquote sneaked in with my commit